helper/funcs.go

```diff
@@ -509,3 +509,10 @@ func Merge[T comparable](a, b T) T {
 	}
 	return a
 }
+
+func MergeListWithDefaults(list []string, defaults []string) []string {
```

Comment:
There's a `Merge` method on all the structs in `nomad/structs/config/ui.go`. We should move this logic and most of the logic you have in `command/agent/http.go` into a `Merge` method on `ContentSecurityPolicy` that gets called by `UIConfig.Merge`.